crates/iceberg/src/spec/schema.rs

```diff
@@ -642,6 +644,204 @@ impl SchemaVisitor for IndexByName {
     }
 }
 
+struct PruneColumn {
+    selected: HashSet<i32>,
+    select_full_types: bool,
+}
+
+/// Visit a schema and returns only the fields selected by id set
+pub fn prune_columns(
+    schema: &Schema,
+    selected: HashSet<i32>,
```

Comment:
```suggestion
    selected: impl IntoIterator<item=i32>,
```
nit: This makes apis easier to use.